tmva/pymva/src/MethodPyKeras.cxx

```diff
@@ -158,10 +160,43 @@ void MethodPyKeras::ProcessOptions() {
    if (fFilenameTrainedModel.IsNull()) {
       fFilenameTrainedModel = GetWeightFileDir() + "/TrainedModel_" + GetName() + ".h5";
    }
+
+   // set here some specific options for Tensorflow backend
+   //  -  when using tensorflow gpu set option to allow memory growth to avoid allocating all memory
+   //  -  set up number of threads for CPU if NumThreads option was specified
+
+   // check first if using tensorflow backend
+   PyRunString("keras_backend_is_tf =  keras.backend.backend() == \"tensorflow\"");
+   PyObject * keras_backend_is_tf = PyDict_GetItemString(fLocalNS,"keras_backend_is_tf");
+   if (keras_backend_is_tf != nullptr && keras_backend_is_tf == Py_True)  {
+      Log() << kINFO << "Using tensorflow backend - setting special config options "  << Endl;
+      PyRunString("import tensorflow as tf");
+      PyRunString("from keras.backend import tensorflow_backend as K");
+      // in case specify number of threads
+      int num_threads = fNumThreads;
+      if (num_threads > 0) {
+         std::cout << "init Pykeras with " << num_threads << std::endl;
+         PyRunString(TString::Format("session_conf = tf.ConfigProto(intra_op_parallelism_threads=%d,inter_op_parallelism_threads=%d)",
+                                     num_threads,num_threads));
+      }
+      else
+         PyRunString("session_conf = tf.ConfigProto()");
+
+      PyRunString("session_conf.gpu_options.allow_growth = True"); // for GPU's (needed for new RTX cards)
```

Comment:
I will just mention this for reference:
As you mentioned during our discussion today this could be extracted to an option.